rts/Sim/Weapons/Weapon.cpp

```diff
@@ -529,6 +531,36 @@ void CWeapon::UpdateSalvo()
 	salvoLeft--;
 	nextSalvo = gs->frameNum + salvoDelay;
 
+	float targetInArc = true;
+	if (weaponDef->maxFireAngle > -1.0f) {
+		const float3 currentTargetDir = (currentTargetPos - aimFromPos).SafeNormalize2D();
+		const float3 simpleWeaponDir = float3(weaponDir).SafeNormalize2D();
+		if (simpleWeaponDir.dot2D(currentTargetDir) < weaponDef->maxFireAngle)
+			targetInArc = false;
+	}
+
+	if (!targetInArc || !CheckAimingAngle()) {
```

Comment:
Isn't the `targetInArc` check redundant?

`CheckAimingAngle` calls `CheckTargetAngleConstraint`
https://github.com/beyond-all-reason/spring/blob/6cefa051663284da0043c12075f11da45808dfda/rts/Sim/Weapons/Weapon.cpp#L351-L361

`CheckTargetAngleConstraint` already checks arc:
https://github.com/beyond-all-reason/spring/blob/6cefa051663284da0043c12075f11da45808dfda/rts/Sim/Weapons/Weapon.cpp#L823-L825

A difference is that yours compares the dir from `aimFromPos` while the existing code checks from unit center. It would be good to replace the existing code with `aimFromPos` though because that would fix subpoint 2 of #341.

Reply:
The difference is more than that, `aimFromPos` moves with the weapon - so if the weapon is on a unit's arm and the unit's torso moves then we get the correct position of the weapon. Also `worldWeaponDir` is derived from the unit's front facing and so, if the turret is rotated by another piece (like a torso) then again, `maxMainDirAngleDif ` will not work correctly.

Reply:
You proved that `maxMainDirAngleDif` is only applicable to turrets that are always on a fixed position on the model. Razorback's weapons move when the torso rotates so `maxMainDirAngleDif` won't correctly constrain the weapon firing arc. So, we need the `targetInArc` check to ensure that burst weapons in this situation cannot fire outside their intended arc nomatter where on the unit they have moved to.

Reply:
I see, it's the other max angle var from the other PR.